google/cloud/bigquery/retry.py

```diff
@@ -29,6 +29,7 @@
     exceptions.BadGateway,
     requests.exceptions.ChunkedEncodingError,
     requests.exceptions.ConnectionError,
+    requests.exceptions.ReadTimeout,
```

Comment:
This is the exception I get if I set the timeout to too small of a value for most requests to complete.

Reply:
Do we need a test for this exception type in `tests/unit/test_retry.py`?

Reply:
Good call. Added.

Looking at the Requests docs, it seems ReadTimeout is just one kind of Timeout error that can happen, so I updated this to the base class and added some tests for other timeouts.